// @esri/calcite-components/dist/esm/sl-aa7861fd.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-components/blob/master/LICENSE.md for details.
 * v1.4.3
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-330adb65.js';
import './index-c7e05e63.js';
import './form-d7359efe.js';
import './dom-1fb94ff6.js';
import './guid-7136974b.js';
import './resources-342e12d1.js';
import './interactive-0eb13600.js';
import './browser-99bc62cf.js';
import './key-8ff0d101.js';
import './label-9d2eb334.js';
import './loadable-8058e300.js';
import './locale-a35648cd.js';
import './observers-95e97fae.js';
import './focusTrapComponent-07adcb3f.js';
import './t9n-fac13394.js';

// Slovenian [sl]
var locale = {
  name: 'sl',
  weekdays: 'nedelja_ponedeljek_torek_sreda_훾etrtek_petek_sobota'.split('_'),
  months: 'januar_februar_marec_april_maj_junij_julij_avgust_september_oktober_november_december'.split('_'),
  weekStart: 1,
  weekdaysShort: 'ned._pon._tor._sre._훾et._pet._sob.'.split('_'),
  monthsShort: 'jan._feb._mar._apr._maj._jun._jul._avg._sep._okt._nov._dec.'.split('_'),
  weekdaysMin: 'ne_po_to_sr_훾e_pe_so'.split('_'),
  ordinal: function ordinal(n) {
    return n + ".";
  },
  formats: {
    LT: 'H:mm',
    LTS: 'H:mm:ss',
    L: 'DD.MM.YYYY',
    LL: 'D. MMMM YYYY',
    LLL: 'D. MMMM YYYY H:mm',
    LLLL: 'dddd, D. MMMM YYYY H:mm'
  },
  relativeTime: {
    future: '훾ez %s',
    past: 'pred %s',
    s: 'nekaj sekund',
    m: 'minuta',
    mm: '%d minut',
    h: 'ura',
    hh: '%d ur',
    d: 'dan',
    dd: '%d dni',
    M: 'mesec',
    MM: '%d mesecev',
    y: 'leto',
    yy: '%d let'
  }
};
dayjs.locale(locale, null, true);

export default locale;
