// @esri/calcite-components/dist/esm/zh-hk-82d06d40.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-components/blob/master/LICENSE.md for details.
 * v1.4.3
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-330adb65.js';
import './index-c7e05e63.js';
import './form-d7359efe.js';
import './dom-1fb94ff6.js';
import './guid-7136974b.js';
import './resources-342e12d1.js';
import './interactive-0eb13600.js';
import './browser-99bc62cf.js';
import './key-8ff0d101.js';
import './label-9d2eb334.js';
import './loadable-8058e300.js';
import './locale-a35648cd.js';
import './observers-95e97fae.js';
import './focusTrapComponent-07adcb3f.js';
import './t9n-fac13394.js';

// Chinese (Hong Kong) [zh-hk]
var locale = {
  name: 'zh-hk',
  months: '一月_二月_三月_四月_五月_六月_七月_八月_九月_十月_十一月_十二月'.split('_'),
  monthsShort: '1月_2月_3月_4月_5月_6月_7月_8月_9月_10月_11月_12月'.split('_'),
  weekdays: '星期日_星期一_星期二_星期三_星期四_星期五_星期六'.split('_'),
  weekdaysShort: '週日_週一_週二_週三_週四_週五_週六'.split('_'),
  weekdaysMin: '日_一_二_三_四_五_六'.split('_'),
  ordinal: function ordinal(number, period) {
    switch (period) {
      case 'W':
        return number + "\u9031";

      default:
        return number + "\u65E5";
    }
  },
  formats: {
    LT: 'HH:mm',
    LTS: 'HH:mm:ss',
    L: 'YYYY/MM/DD',
    LL: 'YYYY年M月D日',
    LLL: 'YYYY年M月D日 HH:mm',
    LLLL: 'YYYY年M月D日dddd HH:mm'
  },
  relativeTime: {
    future: '%s內',
    past: '%s前',
    s: '幾秒',
    m: '一分鐘',
    mm: '%d 分鐘',
    h: '一小時',
    hh: '%d 小時',
    d: '一天',
    dd: '%d 天',
    M: '一個月',
    MM: '%d 個月',
    y: '一年',
    yy: '%d 年'
  }
};
dayjs.locale(locale, null, true);

export default locale;
